### PR TITLE
SNHUT-575, SNHUT-583 - User editing

### PR DESCRIPTION
**Pull request by david-mart**
https://user-images.githubusercontent.com/20728627/182971940-a079b54d-6dac-453e-b44b-6146a7074bb4.mov

### CODE REVIEW

File: apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  async onSubmit() {
+    try {
+      if (this.user) {
+
+        this.failedToUpdateUser = false; //reset if there is an error from previous attempt
+        const userObj = this.userForm.getRawValue() as IUser;
+        this.isLoading = true;
+
+        await this.userService.updateUser(this.user?.id, userObj);
+        console.log('User updated');
```
Comment (joao-sd) [inline apps/toboggan-app/src/app/user/components/edit-user/edit-user.component.ts:135]: Can we remove these console logs?

File: apps/toboggan-app/src/app/shared/components/modal/modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            },
+            style: 'primary',
+          },
+        ]
+      },
+      class: 'gp-modal',
+    });
+    const hideSubscr = this.modal?.onHide?.subscribe(() => {
+      this.hide.emit()
+      console.log('object');
```
Comment (joao-sd) [inline apps/toboggan-app/src/app/shared/components/modal/modal.component.ts:50]: Please remove these console logs

**Review by joao-sd — CHANGES_REQUESTED** [review]
Hey David, that's a great piece of work!
I left a couple of comments regarding some console logs leftovers, but overall, it's excellent.

Thanks

**Review by joao-sd — APPROVED** [review]
LGTM!

**Comment by dmitriy-rodimin-sada** [comment]
Interesting way of using Modal Component from Toboggan UI directly bypassing the Modal Directive. Seems more re-usable than the pattern I came with for Create User. Will try this pattern next time. I assume the instance of the BsModalService would be the same one as injected in the Toboggan UI Allert Banner since it is checking it to hide banner when any modals are shown.

**Review by dmitriy-rodimin-sada — APPROVED** [review]

**Comment by sada-ajith** [comment]
@david-mart  could you please resolve conflicts and merge code  ?

**Pull request merged**